static-assets/components/cstudio-common/common-api.js

```diff
@@ -2481,11 +2481,23 @@ var nodeOpen = false,
                   type: 'CLOSE_SINGLE_FILE_UPLOAD_DIALOG'
                 }
               ]
+            },
+            onClose: {
+              type: 'BATCH_ACTIONS',
+              payload: [
+                {
+                  type: 'DISPATCH_DOM_EVENT',
+                  payload: { id: eventId, type: 'close' }
+                },
+                {
+                  type: 'CLOSE_SINGLE_FILE_UPLOAD_DIALOG'
+                }
+              ]
             }
           }
         });
 
-        CrafterCMSNext.createLegacyCallbackListener(eventId, (result) => {
+        const unsubscribe = CrafterCMSNext.createLegacyCallbackListener(eventId, (result) => {
```

Comment:
`createCustomDocumentEventListener` has auto clean up.
```suggestion
        craftercms.utils.dom.createCustomDocumentEventListener(eventId, (result) => {
```